src/packet.rs

```diff
@@ -1,23 +1,35 @@
 use std::net::SocketAddr;
 
-#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+#[derive(Clone, PartialEq, Eq, Debug, Serialize, Deserialize)]
 pub struct Packet {
-    addr: SocketAddr,
-    seq: Option<u16>,
+    // the address to witch the packet will be send
+    pub addr: SocketAddr,
```

Comment:
I was thinking of creating an overall packet, like in your code previously, because we do not want to waste bytes on ser the socket addr.

Reply:
Good idea but since I saw you had the seq number in Packet I thought you wanted to go for on only One packet. Is that for any reason?

Ill imagine it to be like this:

1. RawPacket (the actual packet that we receive)
2. Packet (wrapper with a socket the user interacts with)

Reply:
For your first question, Yes, I want to break it up into Packet and RawPacket, I just forgot when I started 😄 

For the moment, I would like to leave it out as I want to look into how bincode handles it.